arrows/core/render_mesh_depth_map.h

```diff
@@ -216,6 +216,35 @@ void render_triangle(const vital::vector_2d& v1, const vital::vector_2d& v2, con
 }
 
 
+/// This function renders a triangle filled with label
+/**
+ * \param v1 [in] 2D triangle point
+ * \param v2 [in] 2D triangle point
+ * \param v3 [in] 2D triangle point
+ * \param label [in] value to fill the triangle with
+ * \param img [out] image in which the triangle is rendered
+ */
+template<class T>
+void render_triangle(vital::vector_2d const& v1, vital::vector_2d const& v2,
+                     vital::vector_2d const& v3, T const& label, vital::image& img)
```

Comment:
I'd recommend that `img` be `vital::image_of<T>` as it is in the other render functions.  This forces the caller to make sure they have the correct image type and may make the loop faster by avoiding a cast at each pixel.